blaze/expr/tests/test_literal_expr.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+import pandas as pd
+import pytest
+import re
+
+
+from datashape import dshape
+
+from blaze import discover, transform
+from blaze.expr import data, literal
+from blaze.utils import normalize
+
+
+tdata = (('Alice', 100),
+         ('Bob', 200))
+
+L = [[1, 'Alice',   100],
+     [2, 'Bob',    -200],
+     [3, 'Charlie', 300],
+     [4, 'Denis',   400],
+     [5, 'Edith',  -500]]
+
+l = literal(tdata)
+nl = literal(tdata, name='amounts')
+t = data(tdata, fields=['name', 'amount'])
+
+
+def test_discover_on_data():
+    assert discover(t) == dshape("2 * {name: string, amount: int64}")
+
+
+def test_discover_on_literal():
+    assert discover(l) == dshape("2 * (string, int64)")
+
+
+def test_table_raises_on_inconsistent_inputs():
+    with pytest.raises(ValueError) as excinfo:
+        data(tdata, schema='{name: string, amount: float32}',
+             dshape=dshape("{name: string, amount: float32}"))
+    assert "specify one of schema= or dshape= keyword" in str(excinfo.value)
+
+
+def test_resources():
+    assert t._resources() == {t: t.data}
+
+
+def test_literal_repr():
+    assert "(('Alice', 100), ('Bob', 200))" == repr(l)
+
+
+def test_literal_name_repr():
+    assert "amounts" == repr(nl)
+
+
+def test_data_repr():
+    def replace_name(s):
+        # Account for autogenerated name.
+        return re.sub(r"name='(.*)'", "name='_0'", s)
+    expected = "<'tuple' data; _name='.+', dshape='2 * {name: string, amount: int64}'>" # noqa
```

Comment:
two spaces before noqa, can you also put an empty line after the replace_name function?